storage/mongo/adapter.go

```diff
@@ -53,7 +52,7 @@ func New(url, db, collection string) (storage.Storer, error) {
 	adapter.adaptee = driver
 	adapter.Dispatcher = event.NewDispatcher(0)
 	adapter.Emitter().Use("*", emitter.Void)
-	adapter.addListeners()
+	listeners.AddListeners(adapter.Emitter())
```

Comment:
This line should be not here to avoid coupling as said.

We should attach/plug listeners within an upper tier (eg., the command, a server middleware or similar).

```go
...
ada, err := mongo.New(...)
listeners.AddListeners(ada.Emitter())
...
```